bindings/http/http.go

```diff
@@ -164,9 +174,11 @@ func (h *HTTPSource) Invoke(ctx context.Context, req *bindings.InvokeRequest) (*
 		metadata[key] = strings.Join(values, ", ")
 	}
 
-	// Create an error for non-200 status codes.
-	if resp.StatusCode/100 != 2 {
-		err = fmt.Errorf("received status code %d", resp.StatusCode)
+	// Create an error for non-200 status codes unless suppressed.
+	if treatNon200AsError {
```

Comment:
Can you merge these ifs? `if treatNon200AsError && resp.StatusCode…..`

Reply:
Done.